content/recipes/swagger.md

```diff
@@ -348,52 +412,219 @@ Let's open the browser and verify the generated `Cat` model:
 
 ```typescript
-@ApiUseTags('cats')
 @ApiBearerAuth()
 @Controller('cats')
 export class CatsController {}
 ```
 
-That's how the OpenAPI documentation should look like now:
+Before you run your application, remember to add the security definition to your base document using `DocumentBuilder`:
+
+```typescript
+const options = new DocumentBuilder().addBearerAuth();
+```
+
+#### OAuth2 authentication
+
+In order to enable OAuth2, use `@ApiOAuth2()`.
+
+```typescript
+@ApiOAuth2(['pets:write'])
+@Controller('cats')
+export class CatsController {}
+```
+
+Before you run your application, remember to add the security definition to your base document using `DocumentBuilder`:
 
-<figure><img src="/assets/swagger-auth.gif" /></figure>
+```typescript
+const options = new DocumentBuilder().addOAuth2();
+```
 
 #### File upload
 
-You can enable file upload for a specific method with the `@ApiImplicitFile` decorator together with `@ApiConsumes()`. Here's a full example using [File Upload](/techniques/file-upload) technique:
+You can enable file upload for a specific method with the `@ApiBody` decorator together with `@ApiConsumes()`. Here's a full example using [File Upload](/techniques/file-upload) technique:
 
 ```typescript
 @UseInterceptors(FileInterceptor('file'))
 @ApiConsumes('multipart/form-data')
-@ApiImplicitFile({ name: 'file', required: true, description: 'List of cats' })
+@ApiBody({
+  name: 'file',
+  description: 'List of cats',
+  schema: { type: 'string', format: 'binary' }
+})
 uploadFile(@UploadedFile() file) {}
 ```
 
 #### Decorators
 
-All of the available OpenAPI decorators has an `Api` prefix to be clearly distinguishable from the core decorators. Below is a full list of the exported decorators with a defined use-level (where might be applied).
-
-|                               |                     |
-| ----------------------------- | ------------------- |
-| `@ApiOperation()`             | Method              |
-| `@ApiResponse()`              | Method / Controller |
-| `@ApiProduces()`              | Method / Controller |
-| `@ApiConsumes()`              | Method / Controller |
-| `@ApiBearerAuth()`            | Method / Controller |
-| `@ApiOAuth2Auth()`            | Method / Controller |
-| `@ApiImplicitBody()`          | Method              |
-| `@ApiImplicitParam()`         | Method              |
-| `@ApiImplicitQuery()`         | Method              |
-| `@ApiImplicitHeader()`        | Method              |
-| `@ApiImplicitFile()`          | Method              |
-| `@ApiExcludeEndpoint()`       | Method              |
-| `@ApiUseTags()`               | Method / Controller |
-| `@ApiModelProperty()`         | Model               |
-| `@ApiModelPropertyOptional()` | Model               |
+All of the available OpenAPI decorators has an `Api` prefix to be easily distinguishable from the core decorators. Below is a full list of the exported decorators with a defined use-level (where can be applied).
```

Comment:
All of the available OpenAPI decorators has an `Api` prefix to be easily distinguishable from the core decorators. Below is a full list of the exported decorators with a defined use-level (where can be applied).

->

All of the available OpenAPI decorators have an `Api` prefix to distinguish them from the core decorators. Below is a full list of the exported decorators along with a designation of the level at which the decorator may be applied.